artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java

```diff
@@ -389,7 +389,7 @@ public void sendInternalLarge(CoreMessage message, boolean direct) throws Except
       long id = storageManager.generateID();
       LargeServerMessage largeMessage = storageManager.createLargeMessage(id, message);
 
-      byte[] bytes = new byte[message.getBodyBuffer().writerIndex() - CoreMessage.BODY_OFFSET];
+      byte[] bytes = new byte[message.getBodyBuffer().readableBytes()];
```

Comment:
getReadOnlyBuffer() ???

you would need to cache the bodyBuffer somewhere.. bodyBuffer = message.getReadONlyBuffer();

and only use the bodyBuffer.